bgpd/bgp_route.c

```diff
@@ -188,8 +188,10 @@ static void bgp_info_extra_free(struct bgp_info_extra **extra)
 	if (e->parent) {
 		struct bgp_info *bi = (struct bgp_info *)e->parent;
 
-		if (bi->net)
-			bi->net = bgp_unlock_node((struct bgp_node *)bi->net);
+		if (bi->net) {
+			bgp_unlock_node((struct bgp_node *)bi->net);
+			bi->net = NULL;
```

Comment:
So to bring around a conversation David and I had, this puts back a crash I fixed by putting in this change.  David has a new approach that he is going to try for this issue and we'll take it from there.

Reply:
Also noticed another possible write-after-free here (bi may be freed in bgp_unlock_node...) — new fix pushed here.